pubtools/_quay/operator_pusher.py

```diff
@@ -522,6 +505,112 @@ def build_index_images(self):
                     )
                 )
                 failed_items[id(item)] = True
+        return items_opted_in, failed_items
+
+    def _get_non_fbc_items_for_version(self, items, version, items_opted_in):
+        """Return non fbc items for given ocp version.
+
+        Args:
+            items: List[ContainerPushItem]
+                list of push items
+            version: str
+                ocp version for which items should be returned.
+            items_opted_in: Dict[int, bool]
+                list of push items opted in fbc.
+        Returns List[ContainerPushItem]:
+            list of items not opted in fbc
+        """
+        non_fbc_items = []
+        osev_tuple = tuple([int(x) for x in version.replace("v", "").split(".")])
+        for item in items:
+            if not items_opted_in[id(item)] or (items_opted_in[id(item)] and osev_tuple <= (4, 12)):
+                non_fbc_items.append(item)
+            elif items_opted_in[id(item)] and osev_tuple >= (4, 13):
+                LOG.warning(
+                    "Skipping {i}".format(i=item)
+                    + "from iib build as it's opted in for FBC and targeting OCP version >=4.13"
+                )
+        return non_fbc_items
+
+    def _create_item_groups_for_version(
+        self, non_fbc_items, version, is_hotfix=False, is_prerelease=False
+    ):
+        """Iterate thought non fbc items and group those together based on version.
+
+        Args:
+            non_fbc_items: List[ContainerPushItem]
+                list of items not opted in fbc
+            version: str
+                ocp version
+            is_hotfix: bool
+                flag indicating items are for hotfix push
+            is_prerelease: bool
+                flag indicating items are for prerelease push
+        Returns Dict[str, Dict[str, Any]]:
+            Dictionary of items grouped by ocp version
+        """
+        item_groups = {
+            version: {
+                "items": [],
+                "overwrite": True,
+                "build_tags": [],
+                "destination_tags": [version],
+            }
+        }
+        if is_hotfix or is_prerelease:
+            for item in non_fbc_items:
+                tag_part = (
+                    item.metadata["com.redhat.hotfix"]
+                    if is_hotfix
+                    else item.metadata["com.redhat.prerelease"]
+                )
+                dst_tag = "{0}-{1}-{2}".format(
+                    version,
+                    tag_part,
+                    item.origin.split("-")[1].replace(":", "-"),
+                )
+                item_groups.setdefault(
+                    item.origin,
```

Comment:
The description says that the items are grouped together based on version. But it looks like item_groups can only contain one version, and then it contains item.origin. What is the content of item.origin? Is it a version?

Reply:
yeah, better word would be 'tag'. I'll change it

Reply:
isn't item.origin advisory-id here? so for hotfix/prerelease, the key becomes advisory-id and for others it becomes a version

Follow-up question: Do you generally add tests for this kind of private function used for breaking the workflow into multiple steps? Having a test for this would be useful